nrf_security/Kconfig

```diff
@@ -92,73 +109,57 @@ if CC310_BACKEND
 config HW_CC310_INTERRUPT
 	bool "Use interrupt version of nrf cc310 platform and crypto libraries"
 
-endif
+endif # CC310_BACKEND
 
 config OBERON_BACKEND
 	bool "Use nrf_oberon backend"
 	default y if !(SOC_NRF52840 || (SOC_NRF9160 && TRUSTED_EXECUTION_SECURE))
-	select NRF_CRYPTO_BACKEND_COMBINATION_0 if CC310_BACKEND || MBEDTLS_VANILLA_BACKEND
 	select NRF_OBERON
 	help
-		enable nrf_oberon mbed TLS backend
+	  Enable nrf_oberon mbed TLS backend
 
 config MBEDTLS_VANILLA_BACKEND
-	bool "Use mbed TLS backend"
-	select NRF_CRYPTO_BACKEND_COMBINATION_0 if CC310_BACKEND || OBERON_BACKEND
+	bool "Use mbed TLS vanilla backend"
```

Comment:
Yes.  I haven't fought for this before, but I will fight for this again. 

In a module that provides mbed TLS APIs, there is IMO no better term to accurately describe the fact that said implementations is "unaltered", "uncustomized, "de-facto", "default". 

The nrf_security or "Nordic security" module provides standardized mbed TLS APIs, hence the term "standard" does not cover this nuance. You could say that the mbed TLS API is the "standard".

This is especially clear in terms of the use of "Use mbed TLS backend" which does not give any indication of meaning. Everything in this module is "mbed TLS". Every backend implements an "mbed TLS" backend which provides mbed TLS APIs.

Reply:
There is a difference between `standard mbed TLS` and `mbed TLS API`.
cc310 conforms to `mbed TLS API` but is not `standard mbed TLS`.
What you refer to with vanilla is the `standard mbed TLS` as provided by ARM.

The fact that other backends conforms to the same `mbed TLS API`, does not makes the standard implementation less standard.

Reply:
you could also call it: `arm mbed TLS` or find a 3rd name.
But regardless of the name, the same name should be used in Kconfig menus and the docs, not two different names.

Reply:
> Everything in this module is "mbed TLS"

No, evenrything in this module conforms to the `mbed TLS API`.

Reply:
Working with the tech-writers to find an acceptable term that clearly indicates "unadulterated, root" of the implementation and will work this in for consistency!

Unfortunately "upstream" is a bit hijacked since Zephyr provides its own variant of mbed TLS in-tree and we don't want to inadvertently point to this...

Reply:
> Working with the tech-writers to find an acceptable term that clearly indicates "unadulterated, root" of the implementation and will work this in for consistency!

Could you adjust to current term used in the docs for now, so that all other parts of the PR can go in.
And then take the naming discussion in a dedicated PR, with docs updates ?

> Unfortunately "upstream" is a bit hijacked since Zephyr provides its own variant of mbed TLS in-tree and we don't want to inadvertently point to this...

it's not in-tree upstream, but a fork made into a Zephyr module:
https://github.com/zephyrproject-rtos/mbedtls/

Reply:
Yes. Sorry. In-tree is probably not the right term, but it is a zephyr variant of mbed TLS with alterations (which was my main point)

I'm discussing with tech-writers later today. I have other cleanups in the PR for documentation that merits some restructuring. I can't get CI to pass at the moment in either case, so I'm guessing this won't go in until tomorrow

Reply:
The decision is "Original mbed TLS" as it infers "unaltered" and "origin". 

Kconfig radio-buttons and selections updated. Doc commit updated accordingly